### PR TITLE
Passing actual round leaders' reliability stats to the Engine on round change

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
Here is the "passed by Node" part of https://radixdlt.atlassian.net/browse/SCRY-435 :)

### CODE REVIEW

**Comment by jakrawcz-rdx** [comment]
@dhedey as per our Slack discussion: I have updated the CoreApi to also return this new "proposal history" detail in the validator transaction response, using the indices of validators ordered by their address (which is now the standard ordering of validator set, used throughout Engine, Node and CoreApi).

File: core/src/main/java/com/radixdlt/rev2/REv2StateComputer.java
Comment (dhedey) [inline core/src/main/java/com/radixdlt/rev2/REv2StateComputer.java:254]: So theoretically it's possible for two validators to use the same key. Therefore we should actually use the validator component address here.

File: core-rust-bridge/src/main/java/com/radixdlt/statecomputer/commit/PrepareRequest.java
Comment (dhedey) [inline core-rust-bridge/src/main/java/com/radixdlt/statecomputer/commit/PrepareRequest.java:82]: As per the other comment - we should be using the ComponentAddress rather than the public keys here. (because theoretically two validators in the set could share the same key).

**Review by dhedey — CHANGES_REQUESTED** [review]
Marking as request changes to just have a change to using addresses rather than keys.

I also wonder whether it would be useful to have some kind of end-to-end test of this? Maybe a deterministic test where we drop consensus messages from a particular validator for a single round, and check that validator misses that round in the end of epoch summary event?

**Comment by jakrawcz-rdx** [comment]
> I also wonder whether it would be useful to have some kind of end-to-end test of this?

Maybe on Monday...

**Review by dhedey — APPROVED** [review]
Looks good! Thanks :)

The E2E test will have to wait till after we merge birch anyway.

In the time being, will merge this into my `feature/merge-birch` branch

**Comment by jakrawcz-rdx** [comment]
Let me merge this - as agreed on Slack, an e2e test will come in a separate PR after we merge birch.

**Pull request merged**
